collect_app/src/test/java/org/odk/collect/android/draw/PenColorPickerDialogTest.kt

```diff
@@ -0,0 +1,57 @@
+package org.odk.collect.android.draw
+
+import android.graphics.Color
+import androidx.lifecycle.ViewModelProvider
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.odk.collect.android.R
+import org.odk.collect.android.injection.config.AppDependencyModule
+import org.odk.collect.android.support.CollectHelpers
+import org.odk.collect.fragmentstest.DialogFragmentTest
+import org.odk.collect.fragmentstest.FragmentScenarioLauncherRule
+import org.odk.collect.settings.InMemSettingsProvider
+import org.odk.collect.settings.SettingsProvider
+
+@RunWith(AndroidJUnit4::class)
+class PenColorPickerDialogTest {
+
+    @get:Rule
+    val launcherRule = FragmentScenarioLauncherRule()
+
+    @Test
+    fun `dialog should be cancelable`() {
+        val scenario = launcherRule.launchDialogFragment(PenColorPickerDialog::class.java)
+        scenario.onFragment {
+            assertThat(it.isCancelable, `is`(true))
+        }
+    }
+
+    @Test
+    fun `pen color in view model should be set after clicking ok`() {
+        val factory = PenColorPickerViewModel.Factory(InMemSettingsProvider().getMetaSettings())
```

Comment:
I think you should still use a test double for `PenColorPickerViewModel` here like you were doing before. You can create a fake (using extension of mockito) `PenColorPickerViewModel.Factory` that returns a mock `PenColorPickerViewModel`  and then you check that `setPenColor` is called when you click "OK". 

I think you get clearer tests for Activity/Fragments when you treat them as a "unit" test of the component itself and use doubles for the ViewModel (or any other) layer. Then we don't need to care how the ViewModel itself works.

Reply:
Done.